content/en/blog/2021/external-locality-failover/index.md

```diff
@@ -0,0 +1,217 @@
+---
+title: "Configuring failover for external services"
+description: Learn how to configure locality load balancing and failover for endpoints that are outside of your mesh.
+publishdate: 2021-06-04
+attribution: "Ram Vennam (Solo.io)"
+keywords: [locality,region,failover,Istio,outlier,external]
+---
+
+Istio’s powerful APIs can be used to solve a variety of service mesh use cases. Many users know about it’s strong ingress and east-west capabilities but it also offers many features for egress (outgoing) traffic. This is especially useful when your application needs to talk to an external service - such as a database endpoint provided by a cloud provider. There are often choices of [multiple endpoints](https://docs.aws.amazon.com/general/latest/gr/ddb.html) depending on where your workload is running. You typically want to choose the endpoint closest to your workload for latency reasons, but you may need to configure automatic failover to another endpoint in case things are not working as expected.
+
+Similar to services running inside the service mesh, you can configure Istio to detect outliers and failover to a healthy endpoint, while still being completely transparent to your application. In this example, we’ll use Amazon DynamoDB endpoints and pick a primary region that is the same or close to workloads running in a Google Kubernetes Engine (GKE) cluster. We’ll also configure a failover region.
+
+|Routing|Endpoint|
+|--- |--- |
+|Primary|http://dynamodb.us-east-1.amazonaws.com|
+|Failover|http://dynamodb.us-west-1.amazonaws.com|
+
+![failover](./external-locality-failover.png)
+
+## Define external endpoints using a ServiceEntry
+
+First, determine the location of your workloads.
+
+{{< text bash >}}
+$ kubectl describe node | grep failure-domain.beta.kubernetes.io/region
+                    failure-domain.beta.kubernetes.io/region=us-east1
+                    failure-domain.beta.kubernetes.io/region=us-east1
+{{< /text >}}
+
+As you can see, on my GKE cluster, all of my nodes are running in `us-east1`
+
+Next, apply a Istio ServiceEntry to create a `mydb.com` service that’s backed by the two DynamoDB endpoints.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+ name: external-svc-dns
+spec:
+ hosts:
+ - mydb.com
+ location: MESH_EXTERNAL
+ ports:
+ - number: 80
+   name: http
+   protocol: HTTP
+ resolution: DNS
+ endpoints:
+ - address: dynamodb.us-east-1.amazonaws.com
+   locality: us-east1
+   ports:
+     http: 80
+ - address: dynamodb.us-west-1.amazonaws.com
+   locality: us-west1
+   ports:
+     http: 80
+{{< /text >}}
+
+Let’s deploy a sleep container to runs some tests:
```

Comment:
Personally I like to explain why we use a sleep container, or don't even mention it.  "Deploy a debugging container" or something